src/entity.ts

```diff
@@ -130,86 +128,119 @@ export class Entity {
 		Property$init(prop, this, value);
 	}
 
+	withContext(context: InitializationContext, action: (entity: Entity) => void) {
```

Comment:
@tskimmett I think the code below makes sense. Just curious though, what's the scenario where `withContext` gets called and there's already a context?

Reply:
There isn't a known scenario, but I figured it should be accounted for.

Reply:
Makes sense 👍